Makefile

```diff
@@ -68,18 +68,21 @@ $(TOOLS)/dbotconf: PACKAGE=go.opentelemetry.io/build-tools/dbotconf
 CROSSLINK = $(TOOLS)/crosslink
 $(CROSSLINK): PACKAGE=go.opentelemetry.io/build-tools/crosslink
 
-tools: $(GOLANGCI_LINT) $(MISSPELL) $(GOCOVMERGE) $(STRINGER) $(PORTO) $(MULTIMOD) $(DBOTCONF) $(CROSSLINK)
+GOTMPL = $(TOOLS)/gotmpl
+$(GOTMPL): PACKAGE=go.opentelemetry.io/build-tools/gotmpl
+
+tools: $(GOLANGCI_LINT) $(MISSPELL) $(GOCOVMERGE) $(STRINGER) $(PORTO) $(MULTIMOD) $(DBOTCONF) $(CROSSLINK) $(GOTMPL)
 
 # Build
 
 .PHONY: generate build
 
 generate: $(OTEL_GO_MOD_DIRS:%=generate/%)
 generate/%: DIR=$*
-generate/%: | $(STRINGER)
+generate/%: | $(STRINGER) $(PORTO)
 	@echo "$(GO) generate $(DIR)/..." \
 		&& cd $(DIR) \
-		&& PATH="$(TOOLS):$${PATH}" $(GO) generate ./...
+		&& PATH="$(TOOLS):$${PATH}" $(GO) generate ./... && $(PORTO) -w .
```

Comment:
This approach adds the vanity imports even to `testdata`. See https://github.com/open-telemetry/opentelemetry-go-contrib/pull/3820#discussion_r1196279532

Maybe we could simply do something like following instead?


```make
.PHONY: generate go-generate build

generate: go-generate vanity-import-fix

go-generate: $(OTEL_GO_MOD_DIRS:%=generate/%)
go-generate/%: DIR=$*
go-generate/%: | $(STRINGER) $(PORTO)
	@echo "$(GO) generate $(DIR)/..." \
		&& cd $(DIR) \
		&& PATH="$(TOOLS):$${PATH}" $(GO) generate ./...
```make

Reply:
I'll wait for this ^ change before reviewing.

Reply:
@MrAlias Done: https://github.com/open-telemetry/opentelemetry-go-contrib/pull/3820/commits/80bd9cdf06ff98567dca692b06fafc29e37704db